tools/InstallerSpec/InstallerSpec/Program.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+
+namespace InstallerSpec
+{
+    public class ModuleSpec
+    {
+        public string Name { get; set; }
+        public string FilePath { get; set; }
+        public bool DigitalSignature { get; set; }
+        public bool CopyForInstaller { get; set; }
+        public string Author { get; set; }
+    }
+
+    public class DynamoInstallSpec
+    {
+        [XmlArray]
+        public List<ModuleSpec> Modules { get; set; }
+
+        public static DynamoInstallSpec CreateFromPath(string binPath)
+        {
+            return new DynamoInstallSpec() { Modules = GetModules(new DirectoryInfo(binPath)).ToList() };
+        }
+
+        public static DynamoInstallSpec CreateFromSpecFile(string filePath)
+        {
+            if(string.IsNullOrEmpty(filePath) || !File.Exists(filePath))
+                return null;
+
+            try
+            {
+                var serializer = new XmlSerializer(typeof(DynamoInstallSpec), new[] { typeof(ModuleSpec) });
+                using (var fs = new FileStream(filePath, FileMode.Open, FileAccess.Read))
+                {
+                    return serializer.Deserialize(fs) as DynamoInstallSpec;
+                }
+            }
+            catch (Exception ex) 
+            {
+                Console.WriteLine(string.Format("Failed to Create DynamoInstallSpec from file {0}", filePath));
+                Console.Write(ex.Message);
+            }
+
+            return null;
+        }
+
+        public void Save(string filePath)
+        {
+            var serializer = new XmlSerializer(typeof(DynamoInstallSpec), new[] { typeof(ModuleSpec) });
+            using (var fs = new FileStream(filePath, FileMode.Create, FileAccess.Write))
+            {
+                serializer.Serialize(fs, this);
+            }
+        }
+
+        public static IEnumerable<ModuleSpec> GetModules(DirectoryInfo dir)
+        {
+            foreach (var d in dir.EnumerateDirectories())
+            {
+                var modules = GetModules(d);
+                foreach (var m in modules)
+                {
+                    yield return m;
+                }
+            }
+
+            foreach (FileInfo item in dir.EnumerateFiles())
+            {
+                var info = FileVersionInfo.GetVersionInfo(item.FullName);
+                yield return new ModuleSpec() { Name = item.Name, FilePath = item.FullName.Replace(DynamoCoreVersion.BaseDirectory, @".\"), CopyForInstaller = true, DigitalSignature = Sign(info), Author = info != null ? info.CompanyName : "" };
+            }
```

Comment:
It would be more readable if it's in this form:

```csharp
yield return new ModuleSpec()
{
    Name = item.Name,
    FilePath = item.FullName.Replace(DynamoCoreVersion.BaseDirectory, @".\"),
    CopyForInstaller = true,
    DigitalSignature = Sign(info),
    Author = info != null ? info.CompanyName : ""
};
```

Also, perhaps `Sign` should be renamed to `ShouldBeSigned`?